smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/SmokeTestExtension.java

```diff
@@ -612,4 +617,27 @@ public static Predicate<Envelope> getMetricPredicate(String name) {
       return name.equals(md.getMetrics().get(0).getName());
     };
   }
+
+  public static Predicate<Envelope> getMetricPredicate(
+      String name, String secondPredicate, boolean isRolename) {
+    Objects.requireNonNull(name, "name");
+    Objects.requireNonNull(secondPredicate, "secondPredicate");
+
+    return input -> {
+      if (input == null) {
+        return false;
+      }
+      if (!input.getData().getBaseType().equals("MetricData")
+          || (isRolename && !input.getTags().get("ai.cloud.role").equals(secondPredicate))
+          || (!isRolename && !input.getIKey().equals(secondPredicate))) {
+        return false;
+      }
+      MetricData md = getBaseData(input);
+      boolean isSecondPredicateValid =
+          isRolename
+              ? secondPredicate.equals(input.getTags().get("ai.cloud.role"))
+              : secondPredicate.equals(input.getIKey());
+      return name.equals(md.getMetrics().get(0).getName()) && isSecondPredicateValid;
+    };
+  }
```

Comment:
this seems fairly specific to the one smoke test, can it live in the smoke test itself instead of in the reusable component?

Reply:
it's reused in both connectionstringoverrides and rolenameoverrides.